src/components/button/_button.scss

```diff
@@ -147,6 +153,26 @@
     }
   }
 
+  // Input needs less padding as does not use border-box
+
+  input[type="submit"].govuk-c-button ,
```

Comment:
Will this introduce specificity problems which will make overriding these styles difficult?

Reply:
Right now they need to be ultra specific but really this branch was an experiment to work out why these were behaving weirdly.

Now we know why, I'm expecting all of this to be refactored. As you've mentioned below, perhaps `content-box` could align them all to have the same units.